### PR TITLE
Endelman-Jannink GRM

### LINKED ISSUE #1062: Endelman-Jannink GRM
Add the [Endelman-Jannink](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3484671/) estimator as an option in `genomic_relationship`. This is a straight-forward estimator that doesn't require ancestral allele frequencies. There are existing implementations in R packages `rrBLUP` and `sommer` to test against.

### PR DESCRIPTION
**Pull request by timothymillar**
~~WIP, I need to look into how this interacts with the ancestral_frequencies parameter.~~

- [x] Fixes #1062
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `changelog.rst`

### CODE REVIEW

File: sgkit/stats/grm.py [hunk 1]
```diff
@@ -32,18 +32,60 @@ def _grm_VanRaden(
...
+    G = numerator / denominator
+    return G
+
+
 def genomic_relationship(
     ds: Dataset,
     *,
     call_dosage: Hashable = variables.call_dosage,
-    estimator: Optional[Literal["VanRaden"]] = None,
+    estimator: Optional[Literal["VanRaden", "Endelman-Jannink"]] = None,
```
Comment (jeromekelleher) [inline sgkit/stats/grm.py:72]: I'm a bit nervous about how many times we're using the literal strings here, and the potential for easy-to miss errors ("like 'Endelman-jannink' or something).

Would it make sense to have something like
```
EST_VAN_RADEN = "VanRaden"
EST_ENDELMAN_JANNICK = "Endelman-Jannink"

def genomic_relationship(...
    estimator: Optional[Literal[EST_VAN_RADEN, EST_ENDELMAN_JANNINK]]   
```
and use the constants in the following code? That seems like a reasonable compromise between readability and reducing chances for error, without getting into a tedious global constant declaration thing?

**Review by jeromekelleher — APPROVED** [review]
Looks excellent to me @timothymillar - really clear and concise numerical code :muscle: 

Minor comment on handling of string constants

**Comment by jeromekelleher** [comment]
Would you like to have a look over @tomwhite?

**Comment by timothymillar** [comment]
I'm not quite sure what to do here for suitable testing. The reference implementation is hard coded to use the sample population mean allele frequencies as an approximation of the ancestral frequencies. This is common practice, but it means I can't generate test data for cases when the ancestral frequency is known/differs from the sample mean. We could hard-code this implementation to use the sample mean, but that's at odds with the existing VanRaden implementation.

File: sgkit/stats/grm.py [hunk 2]
```diff
@@ -32,18 +35,62 @@ def _grm_VanRaden(
...
+    m, n = call_dosage.shape
+    ancestral_dosage = ancestral_frequency * ploidy
+    W = call_dosage - ancestral_dosage[:, None]
+    W_mean = da.nanmean(W, axis=0, keepdims=True)
+    Z = W - W_mean
+    S = (Z.T @ Z) / m
+    T = da.diag(S).mean() * da.eye(n)
+    # calculate scaling parameter delta
+    Z2 = Z**2
+    Gama = Z2.T @ Z2 / m  # E&J eq 25
```
Comment (tomwhite) [inline sgkit/stats/grm.py:59]: Should this be `Gamma`?

**Review by tomwhite — APPROVED** [review]
LGTM (I didn't check the maths)

File: sgkit/stats/grm.py [hunk 3]
```diff
@@ -32,18 +35,62 @@ def _grm_VanRaden(
...
+    call_dosage: ArrayLike,
+    ancestral_frequency: ArrayLike,
+    ploidy: int,
+    skipna: bool = False,
+) -> ArrayLike:
+    if skipna:
+        raise NotImplementedError(
+            "The 'skipna' option is not implemented for the '{}' estimator".format(
+                EST_ENDELMAN_JANNINK
+            )
```
[the remark this answers is not in the document]
Comment (timothymillar) [reply sgkit/stats/grm.py:48]: Thanks, I have no objection to pyupgrade

**Comment by timothymillar** [comment]
Ahh, it turns out the Ledoit-Wolf shrinkage estimator it's based on is [implemented in sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.covariance.LedoitWolf.html). So, I've factored out that section as a new function to test independently. This should address the testing concerns I raised above.

**Pull request merged**